Add: 통계 조회, validators 추가

taewooyo opened the pull request:
modified:
        package-lock.json : joi 라이브러리 추가
        modified:   package.json : joi 라이브러리 추가
        modified:   src/api/controllers/group.controller.js : 사용자 통계 로직 추가
        modified:   src/api/models/family.model.js : 사용자 통계 디비 조회
        modified:   src/api/routes/group.router.js : 사용자 통계 라우터 추가
        modified:   src/api/services/group.service.js : 사용자 통계 서비스 로직 추가
        modified:   src/configs/database.config.js : 데이터베이스 timstamp 값 필터링 설정

Add:
        src/api/middlewares/validators/static/ : 통계 유효성 검사 추가

taewooyo committed:
Add: 통계 조회, validators 추가

taewooyo committed:
Add: Validation 추가

taewooyo committed:
Merge branch 'develop' of https://github.com/teamcadi/noogabob-server into taewoo

bbaktaeho reviewed (commented):
```js
if (value.error) {
      res.status(200).json(getApi(false, value.error.details[0].message));
    } else {
      next();
    }
```
에러가 발생했을 때 응답 코드가 200(성공)이라 수정이 필요해 보입니다!
이 부분 따로 에러핸들링으로 상태코드도 보낼 수 있게 만들어놔야겠네여

taewooyo committed:
Update: Validation 오류 에러 핸들링으로 처리

taewooyo commented:
```javascript
const error = new Error(value.error.details[0].message);
error.status = 406;
next(error);
```
변경하였습니다. 맞는지 피드백 부탁드릴게요.

taewooyo committed:
Add: 앨범 기능, 간식 통계

taewooyo commented:
modified:   src/api/controllers/group.controller.js : 이미지 업로드(한개만 업로드가능) 수정필요로 보임
modified:   src/api/middlewares/multer/dog.multer.js : multer jpg 혹은 png만들어오게 설정
modified:   src/api/models/family.model.js : 통계 추가, 이미지 업로드, 가져오기 추가
modified:   src/api/routes/group.router.js : 이미지 가져오는거 추가
modified:   src/api/services/group.service.js : 통계 로직 추가, 앨범 로직 추가
modified:   src/configs/index.js : 이미지 업로드 시 변수명 설정
